### PR TITLE
[EPROD-295] Iceth client

### PR DESCRIPTION
**Pull request by F3kilo**
# Issue ticket

Issue ticket link: https://infinityswap.atlassian.net/browse/EPROD-295
Also, build/deploy/test scripts re-organized.

### Code conventions

- [X] I have performed a self-review of my code
- [X] Every new function is documented
- [X] Object names are auto explicative

### Security

- [X] The PR does not break APIs backward compatibility
- [X] The PR does not break the stable storage backward compatibility

### Testing

- [X] Every function is properly unit tested
- [X] I have added integration tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] IC endpoints are always tested through the `canister_call!` macro

### CODE REVIEW

File: src/eth-signer/test_canister/Cargo.toml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [package]
-name = "ic_sign_test_canister"
+name = "ic-sign-test-canister"
 version = "0.1.0"
```
Comment (ufoscout) [inline src/eth-signer/test_canister/Cargo.toml:3]: version and edition should be inherited from workspace

File: src/iceth-client/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+[package]
+name = "iceth-client"
+version = "0.1.0"
```
Comment (ufoscout) [inline src/iceth-client/Cargo.toml:3]: version and edition should be inherited from workspace

**Review by ufoscout — CHANGES_REQUESTED** [review]

File: src/eth-signer/test_canister/Cargo.toml (hunk 1 above)
Comment (F3kilo) [reply src/eth-signer/test_canister/Cargo.toml:3]: Done

File: src/iceth-client/Cargo.toml (hunk 2 above)
Comment (F3kilo) [reply src/iceth-client/Cargo.toml:3]: Done

File: Cargo.toml [hunk 3]
```diff
@@ -37,6 +39,7 @@ hash-db = "0.16"
 hex = "0.4"
 ic-agent = "0.24"
 ic-canister = { git = "https://github.com/infinity-swap/canister-sdk", package = "ic-canister", tag = "v0.3.x" }
+ic-storage = { git = "https://github.com/infinity-swap/canister-sdk", package = "ic-storage", tag = "v0.3.x" }
```
Comment (ufoscout) [inline Cargo.toml:42]: Could you please replace `infinity-swap` with `bitfinity-network` in all toml files?

Comment (F3kilo) [reply Cargo.toml]: Done

**Review by ufoscout — APPROVED** [review]

**Review by itsyaasir — APPROVED** [review]

File: scripts/build.sh [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+
+export WASM_DIR=.artifact
+mkdir -p $WASM_DIR
+
+echo "Building iceth-client artifacts..."
+./src/iceth-client/scripts/download_iceth.sh
+./src/iceth-client/scripts/build_test_canister.sh
+
+echo "Building eth-signer artifacts..."
+./src/eth-signer/scripts/build_test_canister.sh
```
Comment (GraDKh) [inline scripts/build.sh:11]: [minor] Would be good to fix newlines here and in other places

**Review by GraDKh — APPROVED** [review]

**Pull request merged**